src/OrchardCore/OrchardCore.DisplayManagement/Descriptors/DefaultShapeTableManager.cs

```diff
@@ -19,109 +19,122 @@ namespace OrchardCore.DisplayManagement.Descriptors
     /// </summary>
     public class DefaultShapeTableManager : IShapeTableManager
     {
+        private const string DefaultThemeIdKey = "_ShapeTable";
+
+        // FeatureShapeDescriptors are identical across tenants so they can be reused statically. Each shape table will
+        // create a unique list of these per tenant.
         private static readonly ConcurrentDictionary<string, FeatureShapeDescriptor> _shapeDescriptors = new();
+
         private static readonly object _syncLock = new();
 
-        private readonly IHostEnvironment _hostingEnvironment;
-        private readonly IEnumerable<IShapeTableProvider> _bindingStrategies;
-        private readonly IShellFeaturesManager _shellFeaturesManager;
-        private readonly IExtensionManager _extensionManager;
-        private readonly ITypeFeatureProvider _typeFeatureProvider;
-        private readonly IMemoryCache _memoryCache;
+        // Singleton cache to hold a tenant's theme ShapeTable 
+        private readonly Dictionary<string, ShapeTable> _shapeTableCache;
```

Comment:
Using a singleton dictionary as the cache, like `IMemoryCache`, with the difference that it is indexed by something we don't need to make unique with a prefix since it is only used here.
The dependency is keyed just in case some other component might want to also register the same type.